parsers/US_SPP.py

```diff
@@ -0,0 +1,198 @@
+#!usr/bin/env python3
+
+"""Parser for the Southwest Power Pool area of the United States."""
+
+from dateutil import parser, tz
+from io import StringIO
+from logging import getLogger
+import pandas as pd
+import requests
+
+generation_url = 'https://marketplace.spp.org/public-data-api/gen-mix/asFile'
+
+exchange_url = 'https://marketplace.spp.org/public-data-api/interchange-trend/asFile'
+
+mapping = {'coal': 'coal',
+           'unknown': 'unknown',
+           'Wind': 'wind',
+           'Nuclear': 'nuclear',
+           'Hydro': 'hydro',
+           'Solar': 'solar',
+           'Natural Gas': 'gas',
+           'Diesel Fuel Oil': 'oil',
+           'Waste Disposal Services': 'biomass'
+            }
+
+tie_mapping = {'US-MISO->US-SPP': ['AMRN', 'DPC', 'GRE', 'MDU', 'MEC', 'NSP', 'OTP']}
```

Comment:
I think pylint's preferred style is to have the constants outside the function ALL_CAPS. This actually caught me out when I was reading data_processor() and was wondering where `mapping` came from.